bindings/python/pyproject.toml

```diff
@@ -31,8 +31,30 @@ classifiers = [
   "Programming Language :: Python :: 3.12",
 ]
 
-[project.optional-dependencies]
-test = ["pytest"]
+# Duplicate section required: https://github.com/python-poetry/poetry/pull/9135
```

Comment:
There's an open item to allow poetry to use [project] section instead of [tool.poetry]. I'm exploring using poetry for dependency management because even installing test dependencies is starting to feel a little strange with a simple pyproject.toml file, since we need to maintain a separate list of dependencies in `.github/workflows/bindings_python_ci.yml` as well as in bindings/python/pyproject.toml (there's no way to install just the optional dependencies using pip or maturin, it only supports editable install of the root project in addition to the extra dependencies)

We could also exploring using other python dependency managers as well.

Reply:
Taking my own advice - using pypa/hatch instead: https://hatch.pypa.io/latest/